tls/s2n_connection.c

```diff
@@ -511,6 +521,11 @@ int s2n_connection_set_config(struct s2n_connection *conn, struct s2n_config *co
         }
     }
 
+    /* We do not support handshakes with CLIENT_AUTH and WITH_SESSION_TICKET */
+    if (config->use_tickets && auth_type != S2N_CERT_AUTH_NONE) {
+        config->use_tickets = 0;
```

Comment:
That might be a nasty side effect if somebody used s2n_connection_set_client_auth_type on a single connection, but it turned off session tickets on a whole config, disabling it for all other connecitons.

Reply:
Oh yeah. Will replace 'config->use_tickets = 0;' with S2N_ERROR. Will do the same in s2n_connection_set_client_auth_type too.